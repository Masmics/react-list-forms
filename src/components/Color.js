import React from 'react';
import PropTypes from 'prop-types';

const colorToHex = color => {
  return `${color.toString(16).padStart(2, 0)}`.toUpperCase();
};

const rgbToHex = rgb => { // (rgb) - why parentheses?
  return `${colorToHex(rgb.red)}${colorToHex(rgb.green)}${colorToHex(rgb.blue)}`;
};

function Color({ name, rgb }) { // <---- why deconstructed?
  const hex = rgbToHex(rgb);
  const colorStyle = {
    display: 'inline-block',
    backgroundColor: hex,
    width: '1em',
    height: '1em'
  };

  return (
    <ul>
      <li>
        <li>Color Name: </li>
        <li>{name || hex} <div style={colorStyle}></div></li>
      </li>

      <li>
        <li>Hex: </li>
        <li>{hex} <div style={colorStyle}></div></li>
      </li>

      <li>
        <li>RGB: </li>
        <li>
          <p>Red: {rgb.red}</p>
          <p>Green: {rgb.green}</p>
          <p>Blue: {rgb.blue}</p>
        </li>
      </li>
    </ul>
  );
}

Color.propTypes = {
  name: PropTypes.string,
  rgb: PropTypes.shape({
    red: PropTypes.number.isRequired,
    green: PropTypes.number.isRequired,
    blue: PropTypes.number.isRequired
  })
};

export default Color;
