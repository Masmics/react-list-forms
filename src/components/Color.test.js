import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Color component', () => {
  it('renders one color', () => {
    const wrapper = shallow(<Color name="#" hex="#000000" />);
    expect(wrapper).toMatchSnapshot();
  });
});
