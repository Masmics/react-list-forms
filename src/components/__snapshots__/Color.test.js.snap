// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color component renders one color 1`] = `
<dl>
  <dt>
    Name
  </dt>
  <dd>
    Green
    <div
      style={
        Object {
          "backgroundColor": "#",
          "display": "flexbox",
          "height": "0.85em",
          "width": "0.85em",
        }
      }
    />
  </dd>
  <dt>
    Hex
  </dt>
  <dd>
    #
    <div
      style={
        Object {
          "backgroundColor": "#",
          "display": "flexbox",
          "height": "0.85em",
          "width": "0.85em",
        }
      }
    />
  </dd>
  <dt>
    RGB
  </dt>
  <dd>
    <p>
      Red: 
      NaN
    </p>
    <p>
      Green: 
      NaN
    </p>
    <p>
      Blue: 
      NaN
    </p>
  </dd>
</dl>
`;
